packages/astro/src/core/build/static-build.ts

```diff
@@ -374,12 +369,17 @@ async function cleanStaticOutput(
 	ssrOutputChunkNames: string[]
 ) {
 	const allStaticFiles = new Set();
-	for (const pageData of eachPageData(internals)) {
-		if (pageData.route.prerender && !pageData.hasSharedModules) {
-			const { moduleSpecifier } = pageData;
-			const pageBundleId = internals.pageToBundleMap.get(moduleSpecifier);
-			const entryBundleId = internals.entrySpecifierToBundleMap.get(moduleSpecifier);
+	for (const pageData of internals.pagesByKeys.values()) {
+		const { moduleSpecifier } = pageData;
+		const pageBundleId = internals.pageToBundleMap.get(moduleSpecifier);
+		const entryBundleId = internals.entrySpecifierToBundleMap.get(moduleSpecifier);
+		if (pageData.route.prerender && !pageData.hasSharedModules && allStaticFiles.has(pageBundleId ?? entryBundleId)) {
 			allStaticFiles.add(pageBundleId ?? entryBundleId);
```

Comment:
This doesn't seem right...

Part of the condition is `allStaticFiles.has(pageBundleId ?? entryBundleId)`, which means that adding it to the set will do nothing since it is already in the set.

Also, since the set starts off empty, it seems like `allStaticFiles` will always remain empty as this block will never be reached.

Reply:
Good catch! Should be fixed now.